src/utils/dex/egggroup.js

```diff
@@ -0,0 +1,62 @@
+import { PersonalTable } from '../../../__gamedata';
+
+const EGG_GROUPS = [
+  'None',
+  'Monster',
+  'Water 1',
+  'Bug',
+  'Flying',
+  'Field',
+  'Fairy',
+  'Grass',
+  'Human-Like',
+  'Water 3',
+  'Mineral',
+  'Amorphous',
+  'Water 2',
+  'Ditto',
+  'Dragon',
+  'No Eggs',
+];
+
+const POKEMON_BY_EGG_GROUP = PersonalTable.Personal.reduce(createPokemonByEggGroupMap, []);
+const HIGHEST_EGG_GROUP_ID = 15;
+
+function getEggGroupViaPokemonId(pokemonId = 0) {
+  if (!Number.isInteger(pokemonId) || pokemonId < 0 || pokemonId > PersonalTable.Personal.length)
+    throw new Error(`Bad pokemonId: ${pokemonId}`);
+
+  const pokemonDetails = PersonalTable.Personal[pokemonId];
+  const eggGroup1 = pokemonDetails.egg_group1;
+  const eggGroup2 = pokemonDetails.egg_group2;
+  return eggGroup1 === eggGroup2 ? [eggGroup1] : [eggGroup1, eggGroup2];
```

Comment:
another possible solution: `return new Set([eggGroup1, eggGroup2])` - change if you want to